agrest-cayenne/src/main/java/io/agrest/cayenne/exp/CayenneExpPostProcessor.java

```diff
@@ -113,6 +141,22 @@ public void finishedChild(Expression parentNode, int childIndex, boolean hasMore
                     parentNode.setOperand(childIndex, replacement);
                 }
             }
+            if (parentNode instanceof ASTExists || parentNode instanceof ASTNotExists) {
```

Comment:
Do we need to process both `parentNode` and `childNode` here? Given that we are navigating a tree maybe we could get away with just one?